### PR TITLE
get build_compat_tables.py working

### PR DESCRIPTION
**Pull request by rayrrr**
In pursuit of automatically updating widget implementation status docs, this gets the existing `build_compat_tables.py` script working. This is a script that, when run manually, outputs a `docs/supported_platforms.rst` file showing which components have been implemented per platform to date.

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: contrib/build_compat_tables.py [hunk 1]
```diff
@@ -12,21 +12,23 @@
 A dictionary of components, index is the module name and value is the class name
 """
 COMPONENT_LIST =  [
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column 17: (E222) multiple spaces after operator

File: contrib/build_compat_tables.py [hunk 2]
```diff
@@ -12,21 +12,23 @@
...
     ('Button', 'toga.interface.widgets.button.Button'),
     ('Canvas', 'toga.interface.widgets.canvas.Canvas'),
     ('Icon', 'toga.interface.widgets.icon.Icon'),
-    ('TIBERIUS_ICON', None),
-    ('Image', None),
     ('ImageView', 'toga.interface.widgets.imageview.ImageView'),
-    ('Font', 'toga.interface.font.Font'),
     ('Label', 'toga.interface.widgets.label.Label'),
     ('MultilineTextInput', 'toga.interface.widgets.multilinetextinput.MultilineTextInput'),
     ('NumberInput','toga.interface.widgets.numberinput.NumberInput'),
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column 19: (E231) missing whitespace after ','

File: contrib/build_compat_tables.py [hunk 3]
```diff
@@ -74,7 +76,11 @@ def get_declaration_from_source(text, name="__all__"):
 _maps = dict.fromkeys(COMPONENT_LIST)
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column 1: (E305) expected 2 blank lines after class or function definition, found 1

**Comment by rayrrr** [comment]
screen shot 2018-11-01 at 3 48 01 pm

File: contrib/build_compat_tables.py [hunk 4]
```diff
@@ -71,10 +71,15 @@ def get_declaration_from_source(text, name="__all__"):
                 return ast.literal_eval(node.value)
     raise NameError("name %r was not found"%(name,))
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column 44: (E228) missing whitespace around modulo operator

**Comment by Aareon** [comment]
Should we generally try to avoid %-formatting since `str.format()` is available in all of our target versions? This can cause issues with mistyping variables down the road, or even currently in the codebase. It is also an inconsistency as we use both throughout the project.

File: contrib/build_compat_tables.py [hunk 5]
```diff
@@ -61,20 +61,28 @@
     :width: 32
 """
 
-# Credit http://stackoverflow.com/questions/42195468/how-can-i-inspect-an-attribute-of-a-module-without-importing-it?noredirect=1#comment71553012_42195468
+# Credit http://stackoverflow.com/questions/42195468/
+#        how-can-i-inspect-an-attribute-of-a-module-without-importing-it?noredirect=1#comment71553012_42195468
+
```
Comment (freakboy3742) [inline contrib/build_compat_tables.py]: The long line doesn't need to be split here; splitting means the URL is no longer easily followed. If this is getting flagged by flake8, you can disable the lint check on this single line.

File: contrib/build_compat_tables.py [hunk 6]
```diff
@@ -61,20 +61,28 @@
...
 
 _maps = dict.fromkeys(COMPONENT_LIST)
 
 for module, label in PLATFORM_LIST.items():
-    path = os.path.join('../src', module, 'toga_'+module, '__init__.py')
+    factory_path = os.path.join('../src', module, 'toga_'+module, 'factory.py')
+    if os.path.exists(factory_path):
+        path = factory_path
+    else:
+        path = os.path.join('../src', module, 'toga_'+module, '__init__.py')
```
Comment (freakboy3742) [inline contrib/build_compat_tables.py]: The spacing around the `+` is a little odd here (I'm surprised it isn't picked up by flake8...)

Comment (Aareon) [reply contrib/build_compat_tables.py]: I would suggest using `pathlib` in place of `os.path` as a whole. The API is considerably more Pythonic and easier to reason for most newcomers to the codebase in my opinion.

Comment (freakboy3742) [reply contrib/build_compat_tables.py]: Sure - pathlib is nice, but the *rest* of the codebase isn't using pathlib. Switching directory path mechanisms is a worthwhile change, but it's orthogonal to *this* change, so including that change here wouldn't be appropriate.

A *separate* pull request that ports path usage to pathlib would be welcome, though.

**Comment by Aareon** [comment]
Perhaps I could *start* said merge request, but I may only be able to provide commits for certain backends that I can personally test. I would have to rely on other maintainers to provide the commits and test results for backends like `cocoa` and `iOS`. Aside from that I would be happy to create a PR for this.

File: contrib/build_compat_tables.py (hunk 6 above)
Comment (Aareon) [reply contrib/build_compat_tables.py]: Perhaps I could *start* said merge request, but I may only be able to provide commits for certain backends that I can personally test. I would have to rely on other maintainers to provide the commits and test results for backends like `cocoa` and `iOS`. Aside from that I would be happy to create a PR for this.

Comment (rayrrr) [reply contrib/build_compat_tables.py]: I will ensconce that `+` in spacing today. Thanks!

File: contrib/build_compat_tables.py (hunk 5 above)
Comment (rayrrr) [reply contrib/build_compat_tables.py]: I'll go ahead and fix that up...never disabled a lint for a single line before, sounds like a good skill to have. Thanks!

**Comment by rayrrr** [comment]
@Aareon I'll go ahead and remove the remaining traces of `%`-formatting from this file as well. Doing a commit to address PR feedback today.

File: contrib/build_compat_tables.py (hunk 5 above)
Comment (rayrrr) [reply contrib/build_compat_tables.py]: Actually I did one better and used the "shortened" version of the SO link as per https://meta.stackexchange.com/questions/164194/is-there-a-way-to-shorten-stack-overflow-urls and tested it successfully as pointing to the same SO comment as the original link.

**Comment by rayrrr** [comment]
This should be ready to go. Of note is that nothing here will happen _automatically_ but I would love to see it happen that way. Possibly with GitHub merge hooks or something. Advice welcome.

**Comment by Aareon** [comment]
LGTM 👍🏻

File: contrib/build_compat_tables.py [hunk 7]
```diff
@@ -61,20 +61,26 @@
     :width: 32
 """
 
-# Credit http://stackoverflow.com/questions/42195468/how-can-i-inspect-an-attribute-of-a-module-without-importing-it?noredirect=1#comment71553012_42195468
+# Credit: http://stackoverflow.com/q/42195468#comment71553012_42195468
 def get_declaration_from_source(text, name="__all__"):
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column 1: (E302) expected 2 blank lines, found 1

File: contrib/build_compat_tables.py [hunk 8]
```diff
@@ -55,26 +56,36 @@
...
-.. |no| image:: /_static/no.png
-    :width: 32
+.. Note that X and O symbols can have opposite meanings depending on the culture
+.. so for yes, replace with a check mark and for no, replace with a blank space.
+.. To prevent whitespace-stripping-on-save errors, represent a single blank
+.. space by surrounding it with backslashes.
+.. |yes| replace:: ✔
+.. |no| replace::  \ \
+
 """
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column -16: (W605) invalid escape sequence '\ '

File: contrib/build_compat_tables.py [hunk 9]
```diff
@@ -54,27 +55,37 @@
...
+# space by surrounding it with backslashes.
 _footer = """
-.. |yes| image:: /_static/yes.png
-    :width: 32
-.. |no| image:: /_static/no.png
-    :width: 32
+.. |yes| replace:: ✔
+.. |no| replace::  \ \
+
 """
```
Comment (brutusthebee) [inline contrib/build_compat_tables.py]: At column -16: (W605) invalid escape sequence '\ '

**Comment by rayrrr** [comment]
Gah! I'm trying to be tricky but brutusthebee is having none of it. Going with unicode heavy check mark for "yes," and blank space for "no," in both the text and rendered versions of the output.

**Comment by Aareon** [comment]
Perhaps you can use `# noqa: W605` where appropriate? I don't see that being an issue if it's tested.

**Comment by rayrrr** [comment]
@Aareon thanks for that tip. I will use that in the future if a bot is wrong about something! I'm happy with this in its current state, though. I think "yes or blank" is actually better than "yes or no" given the visualization icon issues surrounding "no."

**Comment by freakboy3742** [comment]
@rayrr Thanks for persisting! @brutusthebee is a fussy bee :-) 

For future reference, if you want to run the checks locally, you can install and run `flake8` from the root of the project directory; that will do the same task that Brutus runs as the first build check.

**Pull request merged**